### PR TITLE
[#21] Fix checkstyle warnings

### LINKED ISSUE #21: Fix checkstyle warnings
once you setup the checkstyle (https://github.com/paypal/NNAnalytics/blob/master/CODESTYLE.md) run the `./gradlew checkstyleMain checkstyleTest` to find out all the checkstyle warnings which needs ti be fixed.

### PR DESCRIPTION
**Pull request by zero45**
Make sure you have checked all steps below.

### GitHub Issue
Fixes #21
Not entirely though -- but it will reduce the WARNINGs from Checkstyle to just things that need to be addressed:
```
LM-SJN-00877839:NNAnalytics pljeliazkov$ ./gradlew clean assemble checkstyleMain 2>&1 > /dev/null | grep  "[ant:checkstyle]" | wc -l
     326
```

### Checklist:

- [ ] This pull request updates the documentation
- [X] This pull request changes the code
- [X] Title of the PR is of format (example) : `[#25] Add Pull Request Template`, where `[#25] is the GitHub issue number`

### What is the purpose of this pull request?
Reduce WARNINGs generated from Checkstyle and simplify issues between Checkstyle and Google Java Style.

### How was this change validated?
Ran all unit tests locally. Ran build with several checks in place:
```
./gradlew clean assemble spotlessJavaCheck licenseMain licenseTest findbugsMain findbugsTest -PhadoopVersion=X.X.X
```

### Commit Guidelines
- [X] My commits all reference GH issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: src/test/java/TestQueryChecker.java [hunk 1]
```diff
@@ -268,21 +286,24 @@ public void testValidQueryChecker() throws Exception {
 
   @Test
   public void testInvalidQueryChecker() throws Exception {
-    HashMap<NNAConstants.SET, HashMap<String, HashMap<String, ArrayList<EnumSet<NNAConstants.FILTER>>>>> setSumTypeFilterConfig = getInvalidCombinations();
+    HashMap<
```
Comment (jhsenjaliya) [inline src/test/java/TestQueryChecker.java]: is this really what checkstyle suggests ? looks untidy to me.

**Comment by r39132** [comment]
To validate this change, I like to ensure it's a stable change. To ensure that, one of the reviewers should be able to checkout this PR and run a reformat in their IDE? The result should not be any additional changes, or is that not guaranteed?

File: src/test/java/TestQueryChecker.java (hunk 1 above)
Comment (zero45) [reply src/test/java/TestQueryChecker.java]: Oh wow good catch. Yeah that looks really untidy.

**Comment by zero45** [comment]
@r39132 -- The main set of changes in 9f2677393f782094afb8bf419c9967d79717925d were generated by running this command:
```
./gradlew spotlessJavaApply
```

I just now ran `Reformat Code` on top of these changes and I do still see differences produced. 
It is mostly resizing the license headers yet there are additional indentation changes as well.

**Comment by jhsenjaliya** [comment]
i ran the code reformat with spotlessJava and got same result, i think there is no good way to reformat these kind of lines, we may have to add custom format (like in checkStyle).

**Comment by r39132** [comment]
I standardized cdh and a few other teams on simple google java style format running on a specific version of idea. We tried a few different approaches before coming to this one. What we found is that other approaches didn’t converge on auto format. One example was that the reformat kept adding new lines to a certain pattern.. so every time someone ran it, there was a diff. Is that what you are seeing?

**Comment by zero45** [comment]
One thing we can do to check if we are getting the same outcome is to compare our Checkstyle results.

Run this:
```
./gradlew clean assemble checkstyleMain 2>&1 > /dev/null | grep  "[ant:checkstyle]" > checkstyle_results.txt
```

And post your checkstyle_results.txt here and we can diff them. I'll post mine shortly.

**Comment by r39132** [comment]
Ok, I see 326 lines in the checkstyle file after checking out this PR. On Master, I see 579, so a 43% reduction. I ran `./gradlew spotlessJavaApply` and then ran `./gradlew clean assemble checkstyleMain 2>&1 > /dev/null | grep  "[ant:checkstyle]" > checkstyle_results.txt` and the number stayed stable at 326.

**Comment by zero45** [comment]
@r39132 I also get the same 326 count as you.

**Comment by zero45** [comment]
I had to `sed` out our workspace directories because it was throwing off the diffs. But once I did I saw that our checkstyle results were matching up. Diff returned empty for our text files and MD5 showed the same checksums for both our files.

Is this enough to move forward with?

File: src/test/java/TestQueryChecker.java (hunk 1 above)
Comment (r39132) [reply src/test/java/TestQueryChecker.java]: The bigger issue here is the crazy level of nesting in this container. Creating some classes to wrap this data structure might be the next step, but I'm not super concerned about the untidiness above per se.

**Comment by r39132** [comment]
Yes. +1.

**Pull request merged**